### PR TITLE
feat(TBD-6295) initial commit for Databricks distribution

### PR DESCRIPTION
**Pull request by StefanLMNG**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -165,6 +166,7 @@
         <module>test/plugins/org.talend.hadoop.distribution.cdh5100.test</module>
         <module>test/plugins/org.talend.hadoop.distribution.cdh5120.test</module>
         <module>test/plugins/org.talend.hadoop.distribution.custom.test</module>
+        <module>main/plugins/org.talend.hadoop.distribution.databricks.test</module>
```
Comment (lqian01) [inline pom.xml]: should be test/plugins/org.talend.hadoop.distribution.databricks.test

File: main/plugins/org.talend.hadoop.distribution.databricks/OSGI-INF/databricks.xml [hunk 2]
```diff
@@ -3,12 +3,9 @@
   <implementation class="org.talend.hadoop.distribution.databricks.DatabricksDistribution"/>
   <service>
     <provide interface="org.talend.hadoop.distribution.component.HadoopComponent"/>
-    <!-- <provide interface="org.talend.hadoop.distribution.component.HiveComponent"/>
     <provide interface="org.talend.hadoop.distribution.component.PigComponent"/> -->
```
Comment (lqian01) [inline main/plugins/org.talend.hadoop.distribution.databricks/OSGI-INF/databricks.xml]: need to remove `-->` in the end of the line

Comment (cmeng-talend) [reply main/plugins/org.talend.hadoop.distribution.databricks/OSGI-INF/databricks.xml]: Seems PigComponent is not implemented by DatabricksDistribution yet.

File: main/plugins/org.talend.hadoop.distribution.databricks/plugin.xml [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                name="HIVEONSPARK-LIB-MRREQUIRED-DATABRICKS">
+             <library id="spark-hive_2.11-2.2.1.jar"/>
+        </libraryNeededGroup>
+        
+        <!-- Spark Azure group -->
+        <libraryNeededGroup
+                description="Hive On SPARK libraries for Databricks"
+                id="SPARK-AZURE-LIB-MRREQUIRED-DATABRICKS"
+                name="SPARK-AZURE-LIB-MRREQUIRED-DATABRICKS">
+             <library id="hadoop-azure-datalake-2.8.1.jar"/>
```
Comment (kcoepeau) [inline main/plugins/org.talend.hadoop.distribution.databricks/plugin.xml]: See TBD-7168

File: main/plugins/org.talend.hadoop.distribution.databricks/src/main/java/org/talend/hadoop/distribution/databricks/DatabricksDistribution.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    }
+	
+    @Override
+    public boolean doSupportAzureBlobStorage() {
+        return true;
+    }
+
+    @Override
+    public boolean doSupportAzureDataLakeStorage() {
+        return true;
```
Comment (kcoepeau) [inline main/plugins/org.talend.hadoop.distribution.databricks/src/main/java/org/talend/hadoop/distribution/databricks/DatabricksDistribution.java]: See TBD-7168

File: main/plugins/org.talend.hadoop.distribution.databricks/src/main/java/org/talend/hadoop/distribution/databricks/modulegroup/node/spark/DatabricksSparkDynamoDBNodeModuleGroup.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+// ============================================================================
+package org.talend.hadoop.distribution.databricks.modulegroup.node.spark;
+
+import java.util.Set;
+
+import org.talend.hadoop.distribution.DistributionModuleGroup;
+import org.talend.hadoop.distribution.databricks.DatabricksConstant;
+import org.talend.hadoop.distribution.utils.ModuleGroupsUtils;
+
+public class DatabricksSparkDynamoDBNodeModuleGroup {
```
Comment (kcoepeau) [inline main/plugins/org.talend.hadoop.distribution.databricks/src/main/java/org/talend/hadoop/distribution/databricks/modulegroup/node/spark/DatabricksSparkDynamoDBNodeModuleGroup.java]: Useless file ?

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/component/HadoopComponent.java [hunk 6]
```diff
@@ -175,7 +175,13 @@
 
     /**
     *
-    * @return true if the distribution is a Qubole cloud distribution
+<<<<<<< HEAD
```
Comment (kcoepeau) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/component/HadoopComponent.java]: merge conflict content in comment

File: test/plugins/org.talend.hadoop.distribution.databricks.test/src/org/talend/hadoop/distribution/databricks/test/DatabricksDistributionTest.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        assertEquals(DatabricksDistribution.DISTRIBUTION_NAME, distribution.getDistribution());
+        assertEquals(DatabricksDistribution.VERSION, distribution.getVersion());
+        assertEquals(EHadoopVersion.HADOOP_2, distribution.getHadoopVersion());
+        assertFalse(distribution.doSupportKerberos());
+
+        assertTrue(distribution.doSupportUseDatanodeHostname());
+        assertFalse(distribution.doSupportGroup());
+
+        assertTrue(((HDFSComponent) distribution).doSupportSequenceFileShortType());
+        assertTrue(((SparkBatchComponent) distribution).getSparkVersions().contains(ESparkVersion.SPARK_2_2));
```
Comment (kcoepeau) [inline test/plugins/org.talend.hadoop.distribution.databricks.test/src/org/talend/hadoop/distribution/databricks/test/DatabricksDistributionTest.java]: Databricks contains SPARK_2_2_1 and not SPARK_2_2.

File: main/plugins/org.talend.hadoop.distribution.databricks/src/main/java/org/talend/hadoop/distribution/databricks/DatabricksDistribution.java (hunk 4 above)
Comment (StefanLMNG) [reply main/plugins/org.talend.hadoop.distribution.databricks/src/main/java/org/talend/hadoop/distribution/databricks/DatabricksDistribution.java]: We  will keep it like that

File: main/plugins/org.talend.hadoop.distribution.databricks/plugin.xml (hunk 3 above)
Comment (StefanLMNG) [reply main/plugins/org.talend.hadoop.distribution.databricks/plugin.xml]: We will keep it like that

**Pull request merged**